src/auth/auth.ts

```diff
@@ -0,0 +1,758 @@
+/*!
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import globals from '../shared/extensionGlobals'
+
+import * as nls from 'vscode-nls'
+const localize = nls.loadMessageBundle()
+
+import * as vscode from 'vscode'
+import * as localizedText from '../shared/localizedText'
+import * as uuid from 'uuid' // TODO: use crypto.randomUUID when C9 is on node16
+import { Credentials } from '@aws-sdk/types'
+import { SsoAccessTokenProvider } from './sso/ssoAccessTokenProvider'
+import { Timeout } from '../shared/utilities/timeoutUtils'
+import { errorCode, isAwsError, isNetworkError, ToolkitError, UnknownError } from '../shared/errors'
+import { getCache } from './sso/cache'
+import { createFactoryFunction, isNonNullable, Mutable } from '../shared/utilities/tsUtils'
+import { builderIdStartUrl, SsoToken } from './sso/model'
+import { SsoClient } from './sso/clients'
+import { getLogger } from '../shared/logger'
+import { CredentialsProviderManager } from './providers/credentialsProviderManager'
+import { asString, CredentialsId, CredentialsProvider, fromString } from './providers/credentials'
+import { once } from '../shared/utilities/functionUtils'
+import { CredentialsSettings } from './credentials/utils'
+import { getCodeCatalystDevEnvId } from '../shared/vscode/env'
+import { partition } from '../shared/utilities/mementos'
+import { SsoCredentialsProvider } from './providers/ssoCredentialsProvider'
+import { AsyncCollection, toCollection } from '../shared/utilities/asyncCollection'
+import { join, toStream } from '../shared/utilities/collectionUtils'
+import { getConfigFilename } from './credentials/sharedCredentialsFile'
+import { SharedCredentialsKeys, StaticProfile, StaticProfileKeyErrorMessage } from './credentials/types'
+import { TempCredentialProvider } from './providers/tempCredentialsProvider'
+import {
+    Connection,
+    ConnectionManager,
+    IamConnection,
+    IamProfile,
+    LinkedIamProfile,
+    Profile,
+    ProfileMetadata,
+    ProfileStore,
+    SsoConnection,
+    SsoProfile,
+    StatefulConnection,
+    StoredProfile,
+    codecatalystScopes,
+    createBuilderIdProfile,
+    hasScopes,
+    isBuilderIdConnection,
+    loadIamProfilesIntoStore,
+    loadLinkedProfilesIntoStore,
+    ssoAccountAccessScopes,
+} from './connection'
```

Comment:
nit: could make sense to use a namespace import e.g. `import * as connection from './connection`. Not sure if the Typescript language server supports this as a refactoring option.

Reply:
Makes sense, I'll keep this in mind since I'll probably be doing some other auth refactors in the future.